girder/models/__init__.py

```diff
@@ -27,10 +27,6 @@
 
 _dbClients = {}
 
-if pymongo.version_tuple < (3,):
-    raise Exception('Your pymongo version (%s) is too old. Please update to '
-                    'pymongo 3.x.' % pymongo.version)
```

Comment:
This isn't related to this PR, let's make this a separate PR.

Reply:
This code will fail when `pymongo` is mocked in a Sphinx-build environment. However, this logic is now obsolete (the correct `pymongo` version will now be ensured by `setup.py` when `girder` is pip-installed, unlike in the past when `girder` was not pip-installed and we just hoped users had used the `requirements.txt` file), so I thought it was safe to remove here. Do you think this still necessitates a separate PR?

BTW, thanks for pushing back on out-of-context changes being slipped into PRs. I think it's good to avoid, I'm just not sure if that's the case here.

Reply:
I didn't realize this change was related to the sphinx build stuff. In light of that, I'd be OK with doing it on this PR, but could you explain how the mocking of packages works? Is that something we configure somewhere, or is it just magically handled for us? And if it's magic, why did this work before and not now? Or was this the cause of all the build failures?

Reply:
The mocking is done by the [`autodoc_mock_imports` option](http://www.sphinx-doc.org/en/stable/ext/autodoc.html#confval-autodoc_mock_imports). This option was not used by Girder in the past, but is a fairly significant part of `sphinx.ext.autodoc`, so I don't think it needs independent documentation here (as we assume developers of `docs/conf.py` understand Sphinx and its common extensions).

Reply:
:shipit: